backtrace-library/src/main/cpp/backtrace-native.cpp

```diff
@@ -48,6 +91,167 @@ JNIEXPORT jint JNI_OnLoad(JavaVM *jvm, void *reserved) {
+    {
+        void *r;
+
+        r = mmap(NULL, BUFFER_SIZE, flags, MAP_SHARED, fd, 0);
+        if (r == MAP_FAILED)
+            abort();
+
+        return r;
+    }
+
+    static void *
+    buffer_create(void)
+    {
+        void *r;
+        int fd;
+
+        fd = bun_memfd_create("_backtrace_buffer", MFD_CLOEXEC);
+        if (fd == -1)
+            abort();
+
+        if (ftruncate(fd, BUFFER_SIZE) == -1)
+            abort();
+
+        buffer_fd = fd;
+
+        r = buffer_reader(buffer_fd, PROT_READ|PROT_WRITE);
+
+        return r;
+    }
+
+    static int
+    request_handler(pid_t tid)
+    {
+        // TODO revisit this logic
+        bool bun_initialized = bun_handle_init(&handle, BUN_BACKEND_LIBUNWINDSTACK);
+        if (!bun_initialized) {
+            return -1;
+        }
+
+        bun_buffer buf = { buffer_child, BUFFER_SIZE };
+
+        auto written = bun_unwind_remote(&handle, &buf, tid);
+
+        (void) written;
+
+        return -1;
+    }
+
+    static void
+    monitor_init(void)
+    {
+        /*
+         * This is called after the parent process has set buffer_fd. Set
+         * a memory mapping to the same descriptor.
+         */
+        buffer_child = (char *)buffer_reader(buffer_fd, PROT_READ | PROT_WRITE);
+        if (buffer_child == NULL)
+            abort();
+
+        pid_t child_pid = getpid();
+
+        memcpy(buffer_child, &child_pid, sizeof(child_pid));
+
+        return;
+    }
+
+    bool LocalUnwindingHandler(int signum, siginfo_t *info, ucontext_t *context)
+    {
+        (void) signum;
+        (void) info;
+        (void) context;
+        size_t bytes_written = bun_unwind(&handle, &buf);
+
+        return false;
+    }
+
+    bool LocalContextUnwindingHandler(int signum, siginfo_t *info, ucontext_t *context)
+    {
+        (void) signum;
+        (void) info;
+        (void) context;
+        size_t bytes_written = bun_unwind_context(&handle, &buf, context);
+
+        return false;
+    }
+
+    bool LocalUnwindingHandlerDumpWithoutCrash(int signum, siginfo_t *info, ucontext_t *context)
+    {
+        (void) signum;
+        (void) info;
+        (void) context;
+
+        thread_local bool flag;
+
+        if (flag == false) {
+            flag = true;
+            size_t bytes_written = bun_unwind(&handle, &buf);
+
+            crashpad::CrashpadClient::DumpWithoutCrash(context);
+
+            return true;
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    bool RemoteUnwindingHandler(int signum, siginfo_t *info, ucontext_t *context)
+    {
+        (void) signum;
+        (void) info;
+        (void) context;
+
+        bcd_emit(&bcd, "1");
+        return false;
+    }
+
+    bool RemoteUnwindingHandlerDumpWithoutCrash(int signum, siginfo_t *info, ucontext_t *context)
+    {
+        (void) signum;
+        (void) info;
+        (void) context;
+
+        thread_local bool flag;
+
+        if (flag == false) {
```

Comment:
this condition could be simplify 
```
if (flag = true) {
return false;
} 
 flag = true;
bcd_emit(&bcd, "1");

crashpad::CrashpadClient::DumpWithoutCrash(context);

return true;
```

Reply:
It does look much better this way, thank you Konrad!